conf/cassandra-env.sh

```diff
@@ -218,55 +218,64 @@ if [ "x$LOCAL_JMX" = "x" ]; then
+  ## Custom auth settings which can be used as alternatives to JMX's out of the box auth utilities.
+  ## JAAS login modules can be used for authentication by uncommenting these two properties.
+  ## Cassandra ships with a LoginModule implementation - org.apache.cassandra.auth.CassandraLoginModule -
+  ## which delegates to the IAuthenticator configured in cassandra.yaml. See the sample JAAS configuration
+  ## file cassandra-jaas.config
+  #JVM_OPTS="$JVM_OPTS -Dcassandra.jmx.remote.login.config=CassandraLogin"
+  #JVM_OPTS="$JVM_OPTS -Djava.security.auth.login.config=$CASSANDRA_CONF/cassandra-jaas.config"
+
+  ## Cassandra also ships with a helper for delegating JMX authz calls to the configured IAuthorizer,
+  ## uncomment this to use it. Requires one of the two authentication options to be enabled
+  #JVM_OPTS="$JVM_OPTS -Dcassandra.jmx.authorizer=org.apache.cassandra.auth.jmx.AuthorizationProxy"
+}
 
-# jmx authentication and authorization options. By default, auth is only
-# activated for remote connections but they can also be enabled for local only JMX
-## Basic file based authn & authz
-JVM_OPTS="$JVM_OPTS -Dcom.sun.management.jmxremote.password.file=/etc/cassandra/jmxremote.password"
-#JVM_OPTS="$JVM_OPTS -Dcom.sun.management.jmxremote.access.file=/etc/cassandra/jmxremote.access"
-## Custom auth settings which can be used as alternatives to JMX's out of the box auth utilities.
-## JAAS login modules can be used for authentication by uncommenting these two properties.
-## Cassandra ships with a LoginModule implementation - org.apache.cassandra.auth.CassandraLoginModule -
-## which delegates to the IAuthenticator configured in cassandra.yaml. See the sample JAAS configuration
-## file cassandra-jaas.config
-#JVM_OPTS="$JVM_OPTS -Dcassandra.jmx.remote.login.config=CassandraLogin"
-#JVM_OPTS="$JVM_OPTS -Djava.security.auth.login.config=$CASSANDRA_CONF/cassandra-jaas.config"
-
-## Cassandra also ships with a helper for delegating JMX authz calls to the configured IAuthorizer,
-## uncomment this to use it. Requires one of the two authentication options to be enabled
-#JVM_OPTS="$JVM_OPTS -Dcassandra.jmx.authorizer=org.apache.cassandra.auth.jmx.AuthorizationProxy"
+# If this function call is commented out, then Cassandra will start with no system properties for JVM set whatsoever.
+# We will be expecting the settings in jmx_server_options and jmx_encryption_options respectively instead.
+# The argument specifies the default port over which Cassandra will be available for JMX connections.
+# If you comment out configure_jmx method call, then JMX_PORT variable will not be set, which means that
+# setting of the port in nodetool which sources this files, will not be done which will parse the default port from
+# cassandra.yaml instead.
+# For security reasons, you should not expose this port to the internet.  Firewall it if needed.
+configure_jmx 7199
```

Comment:
@maulin-vasavada notice the argument of "7199" here. It is set to `JMX_PORT=$1` in that function call which basically creates a variable which we can read in cassandra-env.sh.

So when this is commented out, JMX_PORT will not be set, hence we will try to parse it via cassandra.yaml and if its not there either, we default to 7199 unless a user does not override it on the command line.